tests/mpsc_sender.rs

```diff
@@ -0,0 +1,120 @@
+/// a send fails with an io error.
+#[tokio::test]
+async fn mpsc_sender_clone_closed_error() -> TestResult<()> {
+    tracing_subscriber::fmt::try_init().ok();
+    let (server, client, server_addr) = create_connected_endpoints()?;
+    // accept a single bidi stream on a single connection, then immediately stop it
+    let server = tokio::spawn(async move {
+        let conn = server.accept().await.unwrap().await?;
+        let (_, mut recv) = conn.accept_bi().await?;
+        recv.stop(1u8.into())?;
+        TestResult::Ok(())
+    });
+    let conn = client.connect(server_addr, "localhost")?.await?;
+    let (send, _) = conn.open_bi().await?;
+    let send1 = spsc::Sender::<Vec<u8>>::from(send);
+    let send2 = send1.clone();
+    let send3 = send1.clone();
+    let second_client = tokio::spawn(async move {
+        send2.closed().await;
+    });
+    let third_client = tokio::spawn(async move {
+        // this should fail with an io error, since the stream was stopped
+        loop {
+            match send3.send(vec![1, 2, 3]).await {
+                Err(SendError::Io(e)) if e.kind() == ErrorKind::BrokenPipe => break,
+                _ => {}
+            };
+        }
+    });
+    // send until we get an error because the remote side stopped the stream
+    while send1.send(vec![1, 2, 3]).await.is_ok() {}
+    match send1.send(vec![4, 5, 6]).await {
+        Err(SendError::Io(e)) if e.kind() == ErrorKind::BrokenPipe => {}
+        e => panic!("Expected SendError::Io with kind BrokenPipe, got {:?}", e),
+    };
+    // check that closed signal was received by the second sender
+    second_client.await?;
+    // check that the third sender will get the right kind of io error eventually
+    third_client.await?;
+    // server should finish without errors
+    server.await??;
+    Ok(())
+}
+
+/// Checks that all clones of a `Sender` will get the closed signal as soon as
+/// a send future gets dropped before completing.
+#[tokio::test]
+async fn mpsc_sender_clone_drop_error() -> TestResult<()> {
+    let (server, client, server_addr) = create_connected_endpoints()?;
+    // accept a single bidi stream on a single connection, then read indefinitely
+    // until we get an error
+    let server = tokio::spawn(async move {
+        let conn = server.accept().await.unwrap().await?;
+        let (_, mut recv) = conn.accept_bi().await?;
+        let mut buf = vec![0u8; 1024];
+        while recv.read(&mut buf).await.is_ok() {}
+        TestResult::Ok(())
+    });
+    let conn = client.connect(server_addr, "localhost")?.await?;
+    let (send, _) = conn.open_bi().await?;
+    let send1 = spsc::Sender::<Vec<u8>>::from(send);
+    let send2 = send1.clone();
+    let send3 = send1.clone();
+    let second_client = tokio::spawn(async move {
+        send2.closed().await;
+        // why do I have to do this?
+        // Shouldn't dropping the quinn:SendStream call finish, so the server would get an io error?
+        conn.close(1u8.into(), b"");
```

Comment:
Ah, never mind. I just need to terminate the server also on getting a size of None, which indicates finished.